src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -5896,15 +5896,27 @@ record C1
 
             var comp = CreateCompilation(src);
             comp.VerifyEmitDiagnostics(
-                // (4,35): error CS8870: 'C1.ToString()' cannot be sealed because containing record is not sealed.
+                // (4,35): error CS8652: The feature 'sealed record ToString' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
                 //     public sealed override string ToString() => throw null;
-                Diagnostic(ErrorCode.ERR_SealedAPIInRecord, "ToString").WithArguments("C1.ToString()").WithLocation(4, 35),
-                // (6,8): error CS0239: 'C2.ToString()': cannot override inherited member 'C1.ToString()' because it is sealed
-                // record C2 : C1;
-                Diagnostic(ErrorCode.ERR_CantOverrideSealed, "C2").WithArguments("C2.ToString()", "C1.ToString()").WithLocation(6, 8)
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "ToString").WithArguments("sealed record ToString").WithLocation(4, 35)
                 );
         }
 
+        [Fact]
+        public void ToString_DerivedRecord_BaseHasSealedToString_PreviewSealedRecordToString()
```

Comment:
nit: could be merged into preview test, since same source string (create a compilation and check diagnostics, then repeat with different options).
Please verify execution (using `CompileAndVerify(comp, expectedOutput: "...")`) as well to show that we didn't generate a new ToString.

Reply:
> could be merged into preview test

Do you mean changing the test to a `Theory` instead of `Fact`?

Reply:
> Also with some other ToString methods in base type (parameterlist not empty, return type not string)

I'm unable to introduce a `ToString()` method with a different return type. I get this error:
> error CS8869: 'C2.ToString()' does not override expected method from 'object'.

So I'll just add a method with non-empty parameter list.

(Not sure what this CS8869 error is, btw; I don't get this error if I do the same thing in a class instead of a record. What's the rationale behind this?)

Reply:
> Do you mean changing the test to a Theory instead of Fact?

I had a typo. Meant into "previous" test:
```
string src = "...";
var comp = CreateCompilation(..., Regular9);
comp.VerifyEmitDiagnostics(...);
comp = CreateCompilation(..., RegularPreview);
comp.VerifyEmitDiagnostics();
CompileAndVerify(comp, expectedOutput: "..."); // can verify execution when no error in LangVer=preview
```

Reply:
> I'm unable to introduce a ToString() method with a different return type.

It's possible, but requires jumping through some hoops by defining your own `System.Object` type. See `RecordTests.ObjectEquals_05` for help doing that. The key is to use `CreateEmptyCompilation` to avoid pulling libraries that already have this type.

Reply:
> I had a typo. Meant into "previous" test

Yes, I gathered as much.
Is it OK that I used a theory instead of creating compilations with different options in the same test?

Reply:
> It's possible, but requires jumping through some hoops by defining your own System.Object type. See RecordTests.ObjectEquals_05 for help doing that. The key is to use CreateEmptyCompilation to avoid pulling libraries that already have this type.

Thanks, I'll look into it. Not today, though...

Reply:
Added a test for this scenario (ToString_DerivedRecord_BaseBaseHasSealedToString_And_BaseHasToStringWithDifferentReturnType). I no longer needed to jump through hoops, thanks to the changes related to CS8869